### PR TITLE
Add new usrDir configuration parameter

### LINKED ISSUE #69: It is impossible to use a custom wlp.usr.dir with arquillian
Hi,

##### Issue Overview

by any means, it is really complicated to use a custom `wlp.usr.dir`.

##### Expected Behaviour

Can set WLP_USER_DIR somewow in `pom.xml` and/or not use `WLP_USER_DIR` from environment.

##### Current Behaviour

When a user sets both `WLP_USER_DIR` in his personal environment and `<userDirectory/>` in `liberty-maven-plugin` is used, `liberty-arquillian` cannot be pointed easily to the correct directory.

## Environment
`WLP_USER_DIR=$HOME/.local/share/wlp-usr`

## `pom.xml` 

Setting up liberty for arquillian

```xml
    <wlp.usr.dir>${project.build.directory}/wlp-usr</wlp.usr.dir>
    <liberty.env.wlp.usr.dir>${wlp.usr.dir}</liberty.env.wlp.usr.dir>
```

and the plugins

```xml

      <!-- configure wlp for arquillian -->
      <plugin>
        <groupId>io.openliberty.tools</groupId>
        <artifactId>liberty-maven-plugin</artifactId>
        <version>3.2</version>
        <extensions>true</extensions>
        <!-- Specify configuration, executions for liberty-maven-plugin -->
        <configuration>
          <serverName>ArquillianManagedServer</serverName>

          <assemblyArtifact>
            <groupId>io.openliberty</groupId>
            <artifactId>openliberty-runtime</artifactId>
            <version>[19.0.0.12,)</version>
            <type>zip</type>
          </assemblyArtifact>

          <serverXmlFile>src/test/liberty/config/server.xml</serverXmlFile>
          <userDirectory>${wlp.usr.dir}</userDirectory>

          <bootstrapProperties>
            <default.http.port>${testServerHttpPort}</default.http.port>
            <default.https.port>${testServerHttpsPort}</default.https.port>
          </bootstrapProperties>

          <skipTestServer>true</skipTestServer>
        </configuration>

        <executions>
          <execution>
            <id>create-server-management</id>
            <phase>pre-integration-test</phase>
            <goals>
              <goal>create</goal>
            </goals>
          </execution>

          <execution>
            <id>configure-arquillian-xml</id>
            <phase>pre-integration-test</phase>
            <goals>
              <goal>configure-arquillian</goal>
            </goals>
            <configuration>
              <arquillianProperties>
                <outputToConsole>true</outputToConsole>
              </arquillianProperties>
              <skipIfArquillianXmlExists>true</skipIfArquillianXmlExists>
            </configuration>
          </execution>
        </executions>
      </plugin>

      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-dependency-plugin</artifactId>
        <version>3.1.1</version>
        <executions>
          <execution>
            <id>extract-support-feature</id>
            <phase>pre-integration-test</phase>
            <goals>
              <goal>unpack</goal>
            </goals>
          </execution>
        </executions>
        <configuration>
          <artifactItems>
            <artifactItem>
              <groupId>io.openliberty.arquillian</groupId>
              <artifactId>arquillian-liberty-support</artifactId>
              <version>1.0.6</version>
              <type>zip</type>
              <classifier>feature</classifier>
              <overWrite>true</overWrite>
              <outputDirectory>${wlp.usr.dir}</outputDirectory>
            </artifactItem>
          </artifactItems>
        </configuration>
      </plugin>
```

## Expected outcome

Arquillian will start the server via `WLPManagedContainer.java`

## Actual outcome

Line breaks are original.

```
Mar 25, 2020 4:32:56 PM io.openliberty.arquillian.managed.WLPManagedContainer start
FINER: Starting server with command: [$HOME/svn/project/web/rest-v1/target/liberty/wlp/bin/server, run, ArquillianManagedServer]



Server ArquillianManagedServer d
oes not exist.
```

## Known workarounds

None. At least no easy one. 

* I tried `liberty.env.wlp.usr.dir`, but it will only set a server-specific `server.env` file.
* I tried multiple `arquillianProperties`, but even the java options have no effect because as you can see [WLPManagedContainer.java](https://github.com/OpenLiberty/liberty-arquillian/blob/master/liberty-managed/src/main/java/io/openliberty/arquillian/managed/WLPManagedContainer.java) will only read `WLP_USER_DIR` from either a system-wide `server.env` or a environment variable.
* I can unset the global `WLP_USER_DIR` by setting the jvm option `-Dwlp.ignore.user.dir.from.env=true` via `arquillianProperties` in `pom.xml`, but this will make liberty look in `${project.build.directory}/liberty/usr` again, which is kind of correct, I guess.

Working workaround if, and only if `maven-resoures-plugin` is defined AFTER the `liberty-maven-plugin`:

```xml
<plugin>
        <artifactId>maven-resources-plugin</artifactId>
        <executions>
          <execution>
            <id>copy-global-server.env</id>
            <phase>pre-integration-test</phase>
            <goals>
              <goal>copy-resources</goal>
            </goals>
            <configuration>
              <outputDirectory>${project.build.directory}/liberty/wlp/etc</outputDirectory>
              <resources>
                <resource>
                  <directory>${basedir}/src/test/liberty/etc</directory>
                  <filtering>true</filtering>
                </resource>
              </resources>
            </configuration>
          </execution>
        </executions>
      </plugin>
```

## Proposed solution

`WLPManagedContainer` cannot look up the `pom.xml` file for the other plugin's `userDirectory` property.
So I think the next best thing would be to have the `maven-liberty-plugin` write  a system-wide `server.env` file in `${project.build.directory}/liberty/etc/server.env`.

### PR DESCRIPTION
**Pull request by chyt**
Fixes #69

### CODE REVIEW

File: liberty-managed/src/main/java/io/openliberty/arquillian/managed/WLPManagedContainer.java [hunk 1]
```diff
@@ -1405,10 +1409,17 @@ public String run() {
...
-         usrDir = containerConfiguration.getWlpHome() + "/usr/";
+      String usrDir;
+      if(containerConfiguration.getUsrDir() != null) {
+         usrDir = containerConfiguration.getUsrDir();
+      }
+      else {
+         usrDir = getLibertyEnvVar(WLP_USER_DIR);
+         if (usrDir == null) {
+            usrDir = containerConfiguration.getWlpHome() + "/usr/";
+         }
```
Comment (cherylking) [inline liberty-managed/src/main/java/io/openliberty/arquillian/managed/WLPManagedContainer.java:1420]: Could simplify this a tiny bit:
String usrDir = containerConfiguration.getUsrDir();
if (usrDir == null) {
    usrDir = getLibertyEnvVar(WLP_USER_DIR);
....

File: liberty-managed/src/main/java/io/openliberty/arquillian/managed/WLPManagedContainer.java [hunk 2]
```diff
@@ -1437,7 +1445,7 @@ private String getServerOutputDir() throws IOException {
     * @throws IOException
     */
    private String getServerConfigDir() throws IOException {
-	   String serverConfigDir = getWlpUsrDir() + "servers/" + containerConfiguration.getServerName();
+	   String serverConfigDir = getWlpUsrDir() + "/servers/" + containerConfiguration.getServerName();
```
Comment (bmhm) [inline liberty-managed/src/main/java/io/openliberty/arquillian/managed/WLPManagedContainer.java:1448]: Your PR seems to superseed mine?

https://github.com/OpenLiberty/liberty-arquillian/pull/68

Comment (cherylking) [reply liberty-managed/src/main/java/io/openliberty/arquillian/managed/WLPManagedContainer.java:1448]: Yes, it seems so. Sorry for the inconvenience.

Comment (bmhm) [reply liberty-managed/src/main/java/io/openliberty/arquillian/managed/WLPManagedContainer.java:1448]: No worries, I'll just close mine.

**Pull request merged**